SAMv1.tex

```diff
@@ -500,11 +500,33 @@ \subsection{The alignment section: mandatory fields}\label{sec:alnrecord}
   {\sf RNEXT} and bit 0x20.
 \item {\sf TLEN}: signed observed Template LENgth. If all segments are
   mapped to the same reference, the unsigned observed template length
-  equals the number of bases from the leftmost mapped base to the
-  rightmost mapped base. The leftmost segment has a plus sign and the
-  rightmost has a minus sign. The sign of segments in the middle is
-  undefined. It is set as 0 for single-segment template or when the
-  information is unavailable.
+  equals the distance between the mapped end of the template and the
+  mapped start of the template, inclusively.\footnote{Thus a segment
+    aligning in the forward direction at base 100 for length 50 and a
+    segment aligning in the reverse direction at base 200 for length
+    50 indicate the template covers bases 100 to 250 and has length
+    151.}
+  Note that mapped base is defined to be one that aligns to the
+  reference as described by CIGAR, hence exclude soft-clipped bases.
+  The TLEN field is positive for leftmost segment of the template,
+  negative for the rightmost and the sign for any middle segment is
```

Comment:
what about two reads with the same exact alignment? which is negative and which positive?

Reply:
I thought the spec already stated this case gives an arbitrary choice, but I can't see it.  Maybe I'm remembering something else.

We should add something that states leftmost and rightmost can be arbitrarily chosen when they are both equal.